File: iot-simulator/src/main/java/iot/day/Temperature.java
package iot.day;

public class Temperature {

    private String deviceId;

    private double temperature;

    private long timestamp;

    public Temperature() {
    }

    public Temperature(String deviceId, double temperature, long timestamp) {
        this.deviceId = deviceId;
        this.temperature = temperature;
        this.timestamp = timestamp;
    }

    public String getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    public double getTemperature() {
        return temperature;
    }

    public void setTemperature(double temperature) {
        this.temperature = temperature;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "Temperature{" +
                "deviceId='" + deviceId + '\'' +
                ", temperature=" + temperature +
                ", timestamp=" + timestamp +
                '}';
    }

}

File: iot-analysis/src/main/scala/iot/day/Temperature.scala
package iot.day

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule

case class Temperature(deviceId: String, temperature: Double, timestamp: Long) {

  def toJson: String = Temperature.toJson(this)

}

object Temperature {

  private lazy val objectMapper = new ObjectMapper().registerModule(DefaultScalaModule)

  def fromJson(json: String): Temperature = objectMapper.readValue(json, classOf[Temperature])

  def toJson(event: Temperature): String = objectMapper.writeValueAsString(event)

}
